src/storage/DataAccessorBasedStore.ts

```diff
@@ -196,7 +196,7 @@ export class DataAccessorBasedStore implements ResourceStore {
       throw new ConflictHttpError(`${identifier.path} conflicts with existing path ${oldMetadata.identifier.value}`);
     }
 
-    const isContainer = this.isNewContainer(representation.metadata, identifier.path);
+    const isContainer = this.isContainerType(representation.metadata);
     // Solid, §3.1: "Paths ending with a slash denote a container resource."
     // https://solid.github.io/specification/protocol#uri-slash-semantics
     if (isContainer !== isContainerIdentifier(identifier)) {
```

Comment:
```suggestion
    // Solid, §3.1: "Paths ending with a slash denote a container resource."
    // https://solid.github.io/specification/protocol#uri-slash-semantics
    const isContainer = isContainerIdentifier(identifier);
    if (!isContainer && this.isContainerType(representation.metadata)) {
```
We probably still had a communication error somewhere, but this change fixes all your integration issues.